.github/workflows/fast_testing.yml

```diff
@@ -0,0 +1,47 @@
+name: fast_testing
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  linux:
+    # We want to run on external PRs, but not on our own internal
+    # PRs as they'll be run by the push to the branch.
+    #
+    # The main trick is described here:
+    # https://github.com/Dart-Code/Dart-Code/pull/2375
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+
+    strategy:
+      fail-fast: false
+      matrix:
+        tarantool:
+          # TODO: Add 2.9 version after https://github.com/tarantool/vshard/issues/294
+          # is resolved.
+          - '1.10'
+          - '2.7'
+          - '2.8'
+          #- '2.9'
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install tarantool ${{ matrix.tarantool }}
+        uses: tarantool/setup-tarantool@v1
+        with:
+          tarantool-version: ${{ matrix.tarantool }}
+
+      - name: Clone the module
+        uses: actions/checkout@v2
+        # Enable recursive submodules checkout as test-run git module is used
+        # for running tests.
+        with:
+          fetch-depth: 0
+          submodules: recursive
```

Comment:
Cited my comment above:

> I would also leave a comment why we need the `fetch-depth` option.

Your comment is about the `submodules` option. But the `fetch-depth` one is less obvious as for me (because you need to know how the upgrade testing works, not just looks, whether there are submodules).

Reply:
Sorry, I misunderstood you. Fixed.